src/orders/service/orderDetails.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from '@nestjs/common';
+import OrderDetailInDTO from '../dto/orderDetail.indto';
+import { StocksService } from 'src/products/service/stocks.service';
+import OrderDetailsRepository from '../repository/orderDetails.repository';
+import OrderDetail from '../domain/orderDetail.domain';
+
+@Injectable()
+export class OrderDetailsService {
+  constructor(
+    private orderDetailsRepository: OrderDetailsRepository,
+    private stocksService: StocksService,
+  ) {}
+
+  async validateOrderDetails(
+    orderDetails: OrderDetailInDTO[],
+  ): Promise<boolean> {
+    let ordersValid = true;
+    await Promise.all(
+      orderDetails.map(async (orderDetail) => {
+        const stock = await this.stocksService.getStock(
+          orderDetail.productId,
+          orderDetail.shippedFrom,
+        );
+        if (stock.quantity < orderDetail.quantity) {
+          ordersValid = false;
+        }
+      }),
+    );
+    return ordersValid;
+  }
+
+  async decreaseStock(orderDetails: OrderDetailInDTO[]): Promise<void> {
+    await Promise.all(
```

Comment:
Nice use of `Promise.all`!

Probably this method could be moved to `StockService` and called from `OrderService`.

Reply:
done